rust/src/treefile.rs

```diff
@@ -509,7 +535,11 @@ struct TreeComposeConfig {
     #[serde(skip_serializing_if = "Option::is_none")]
     selinux: Option<bool>,
     #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(rename = "gpg-key")]
     gpg_key: Option<String>,
+    #[serde(skip_serializing)]
+    #[serde(rename = "gpg_key")]
+    gpg_key_legacy: Option<String>,
```

Comment:
I wonder if it would work to do:

```
diff --git a/rust/src/treefile.rs b/rust/src/treefile.rs
index e1d20083..6bed601a 100644
--- a/rust/src/treefile.rs
+++ b/rust/src/treefile.rs
@@ -492,6 +492,11 @@ struct Rojig {
     description: Option<String>,
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+struct LegacyTreeComposeConfig {
+    /* Define legacy keys here */
+}
+
 // Because of how we handle includes, *everything* here has to be
 // Option<T>.  The defaults live in the code (e.g. machineid-compat defaults
 // to `true`).
@@ -618,6 +623,9 @@ struct TreeComposeConfig {
     #[serde(skip_serializing_if = "Option::is_none")]
     #[serde(rename = "remove-from-packages")]
     remove_from_packages: Option<Vec<Vec<String>>>,
+
+    #[serde(flatten)]
+    legacy: LegacyTreeComposeConfig,
 }
 
 #[derive(Serialize, Deserialize, Debug)]
```

Reply:
See how we use `serde(flatten)` for `struct PermissiveTreeComposeConfig` and `struct StrictTreeComposeConfig`.